charts/graphprotocol-node/templates/deployment.yaml

```diff
@@ -0,0 +1,107 @@
+  {{- if eq .Values.role "index-node" -}}
+  replicas: 1
+  strategy:
+    type: Recreate
+  {{- else -}}
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: RollingUpdate
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "graphprotocol-node.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/configmap.yaml: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- with .Values.podAnnotations }}
+        {{-   toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "graphprotocol-node.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: BLOCK_INGESTOR
+              value: {{ .Values.blockIngestorNodeId | quote }}
+            - name: node_id
+              {{/* Index (ingest) node_id should equal BLOCK_INGESTOR env value to work in this mode */}}
+              {{- if eq .Values.role "index-node" -}}
+              value: {{ .Values.blockIngestorNodeId | quote}}
+              {{- else -}}
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+              {{- end }}
+            - name: node_role
+              value: {{ .Values.role | quote }}
+            - name: GRAPH_KILL_IF_UNRESPONSIVE
+              value: "true"
+            - name: postgres_db
+              value: {{ required "Postgres name wasn't specified" .Values.postgres.db | quote }}
+            - name: postgres_host
+              value: {{ required "Postgres host wasn't specified" .Values.postgres.host | quote }}
+            - name: ipfs
+              value: {{ required "IPFS address wasn't specified" .Values.config.ipfs | quote }}
+          envFrom:
+            - secretRef:
+                name: {{ include "graphprotocol-node.fullname" . }}
+          ports:
+            - name: metrics 
+              containerPort: 8040
+              protocol: TCP
+            - name: json-rpc 
+              containerPort: 8020
+              protocol: TCP
+            - name: graphql
+              containerPort: 8000
+              protocol: TCP
+            - name: graphql-ws
+              containerPort: 8001
+              protocol: TCP
+            {{- if eq .Values.role "index-node" }}
+            - name: index
+              containerPort: 8030
+              protocol: TCP
+            {{- end }}
+          livenessProbe:
```

Comment:
For now we can leave it as it is now.
But I'm thinking that If `json-rpc` gets opened only after node is successfully launched, then we could use  just`tcpSocket` for it. 

Feel free to close conversation, as probes can be adjusted later

Reply:
I think tcpSocket is not good enough cause during probe you application not involved, so it can be deadly hanged up and probe would be still succesfull.

So let's return later to this #4